### PR TITLE
REG-451: added typescript

### PR DESCRIPTION
**Pull request by nickshatilo**
**What has been done:**
* Added typescript, `typechain` package
* Rewrote all scripts, deployment scripts, sandbox, tests to typescript
* Added typescript to eslint

### 1. Contracts versioning
- [x] Make sure that the `patch` version of the contracts is increased if changes have been made to the `UNSRegistry`, `MintingManager` or `ProxyReader` contracts.
- [ ] Make sure that the `minor` version of the contracts is increased if breaking changes have been made to the `UNSRegistry`, `MintingManager` or `ProxyReader` contracts. It includes changes of interfaces.
### 2. Contracts licensing
- [x] Make sure that no **SPDX-License-Identifier** defined in contracts.
- [ ] Make sure that the **header** is added to the new contract files. 
  ```
  // @author Unstoppable Domains, Inc.
  // @date {Month} {Day}(ordinal), {Year}
  ```
### 3. Coverage
- [x] Make sure that the coverage of contracts has not decreased and strive **100%**
### 4. Configs versioning
- [x] Make sure that the version of `uns-config.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `resolver-keys.json` is increased if changes have been made to the config.
### 5. Sandbox
- [x] Make sure that the Sandbox is updated if changes have been made to contracts or deployment configuration.
  ```
  yarn rebuild:sandbox
  ```
### 6. Package versioning
- [x] Make sure that the `patch` version of package is increased if valuable changes have been made to the package. It includes contracts update, configs update, etc.
- [ ] Make sure that the `major.minor` version of package is synced with version of `UNSRegistry` contract.
- [ ] Make sure `tag` was released for the new version of the package in order to be able to add the package as a dependency.
  ```
  git tag v{version}
  git push origin --tags
  ```
- [x] Make sure that the `CHANGELOG` is updated with short description for the new version. 
### 7. Code review
- [ ] `resolver-keys.json` code review is required from **DevTools** team
- [x] For all other changes code review is required from **Registry** team

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,24 +32,40 @@
     "analyze": "slither . --hardhat-artifacts-directory ./.artifacts --sarif slither.sarif --config-file .slither.json"
   },
   "devDependencies": {
+    "@ethereumjs/tx": "^4.0.1",
```
Comment (bogdan) [inline package.json:35]: Why did this lib become explicitly defined?

File: sandbox/index.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    return new Sandbox(service, options);
+  }
+
+  ganacheService: GanacheService;
+  options: SandboxOptions;
+  provider: EthereumProvider;
+  private _snapshotId?: string;
+  version: string;
+
+  accounts: Map<string, SandboxAccount>;
```
Comment (bogdan) [inline sandbox/index.ts:122]: @aquiladev please tell us which of these properties would really need to be public

File: sandbox/index.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+  async _snapshot () {
+    return await this.provider.send('evm_snapshot');
+  }
+
+  async _revert (snapshotId: string) {
+    return await this.provider.send('evm_revert', [snapshotId]);
+  }
+
+  _getAccounts (serverOptions: ServerOptions<'ethereum'>) {
```
Comment (bogdan) [inline sandbox/index.ts:192]: I believe we now can use `private` instead of `_` prefix convention.

Comment (bogdan) [reply sandbox/index.ts:192]: `private` methods can be called and tested like `object["private_method"]()`. This is helpful for open source code.

File: package.json (hunk 1 above)
Comment (nickshatilo) [reply package.json:35]: We have a number of `@ethereum/tx` packages with different versions, and they differ in type declarations, so sometimes when we assume there will be one type, we are able to only include another. So a global version setting for the whole package fix this situation

File: sandbox/index.ts (hunk 2 above)
Comment (nickshatilo) [reply sandbox/index.ts:122]: Most of them are used in public, especially in the `tasks.ts` file. I'll recheck each one

File: sandbox/index.ts (hunk 3 above)
Comment (helfi92) [reply sandbox/index.ts:192]: Ah, great. I didn't know about this feature. Thanks for sharing 🥇

**Comment by aquiladev** [comment]
cant run tests locally
```
yarn
yarn compile
yarn test
```
Screen Shot 2022-11-02 at 9 23 47 AM

**Review by bogdan — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -40,5 +40,7 @@ cache
 .sandbox
 .deployer
 
-slither.sarif
```
Comment (aquiladev) [inline .gitignore:43]: it is better to revert

File: uns-config.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-    "version": "0.6.1",
+    "version": "0.6.2",
```
Comment (aquiladev) [inline uns-config.json:2]: no need to change the version when file didn't changed

**Review by aquiladev — APPROVED** [review]

**Pull request merged**